ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java

```diff
@@ -164,10 +166,13 @@ public void startup() throws InstantiationException {
             + "is not present in the clustermap. Failing to start the datanode");
       }
 
+      HelixAccountServiceFactory helixAccountServiceFactory = new HelixAccountServiceFactory(properties, registry);
```

Comment:
To allow for plugable implementations of AccountService, it would be best to use reflection to instantiate the account service. Doing so will allow us to start ambry-server in environments where zookeeper is not running, such as the quickstart on the github readme. See https://github.com/linkedin/ambry/blob/master/ambry-rest/src/main/java/com/github/ambry/rest/RestServer.java#L171 for an example of how to use Utils.getObj to instantiate AccountService.

Reply:
Thanks for the suggestion. Updated.